### PR TITLE
fix: Invalid default VCS Provider; add tests

### LINKED ISSUE #898: Failing on empty vcs-provider
Hi,

This happens with recently released version 0.10.7 and wasn't happenning with 0.10.5. When vcs-provider is unset we get following error

```
{"level":"error","error":"could not validate vcs-provider: invalid value ; must be one of: github","time":"2026-08-26T12:29:26Z"}
```

But according to https://github.com/terralist/terralist/blob/master/docs/configuration.md it is not required. Fix is to either set it by default to github or accept it unset.

### PR DESCRIPTION
**Pull request by abborg**
Fix scenario where unset vcs-provider flag results in an error due to the flag value being an invalid choice. This is done by ignoring an unset string flag if not required when the value is an empty string. Previously, during flag value validation, string flags would be checked for choice validity even if they were unset and not required. Now, unset string flags with choices do not result in errors if not required.

This PR also adds tests and fixes some suspected bugs in flag parsing. These bugs relate to scenarios where the flag value is set to the zero value for the type (i.e. false for bool, empty string for string and path, and zero for int). 

Fixes #898

### CODE REVIEW

File: pkg/cli/int_flag.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ func (t *IntFlag) Set(value any) error {
 		if v == 0 {
 			if v != t.DefaultValue {
 				t.Value = v
+				t.isSet = true
```
Comment (abborg) [inline pkg/cli/int_flag.go]: This behavior of allowing the zero value for the type to override the default value set for the flag, but also not setting `isSet` to true seems unintentional. If it is intentional, these changes (and their associated tests), can be reverted.

File: pkg/cli/path_flag.go [hunk 2]
```diff
@@ -45,6 +45,7 @@ func (t *PathFlag) Set(value any) error {
 		if v == "" {
 			if v != t.DefaultValue {
 				t.Value = v
+				t.isSet = true
```
Comment (abborg) [inline pkg/cli/path_flag.go]: See the related comment on int_flag.go

File: pkg/cli/bool_flag.go [hunk 3]
```diff
@@ -52,6 +52,7 @@ func (t *BoolFlag) Set(value any) error {
 		if !v {
 			if v != t.DefaultValue {
 				t.Value = v
+				t.isSet = true
```
Comment (abborg) [inline pkg/cli/bool_flag.go]: See the related comment on int_flag.go

File: cmd/terralist/server/server.go [hunk 4]
```diff
@@ -501,11 +501,6 @@ func (s *Command) run() error {
 			AppPrivateKeyPath: flags[GitHubAppPrivateKeyPathFlag].(*cli.StringFlag).Value,
 			BaseURL:           flags[GitHubBaseURLFlag].(*cli.StringFlag).Value,
 		})
-	default:
-		raw := flags[VcsProviderFlag].(*cli.StringFlag).Value //nolint:forcetypeassert
-		if raw != "" {
-			return fmt.Errorf("invalid VCS provider: %s", raw)
-		}
```
Comment (abborg) [inline cmd/terralist/server/server.go:508]: Removed as this unnecessarily adds validation that is actually better captured in the flag choice validation.

File: pkg/cli/string_flag.go [hunk 5]
```diff
@@ -44,6 +44,7 @@ func (t *StringFlag) Set(value any) error {
 		if v == "" {
 			if v != t.DefaultValue {
 				t.Value = v
+				t.isSet = true
```
Comment (abborg) [inline pkg/cli/string_flag.go]: See the related comment on int_flag.go

File: pkg/cli/string_flag.go [hunk 6]
```diff
@@ -85,6 +86,10 @@ func (t *StringFlag) Validate() error {
 		return fmt.Errorf("required but not set")
 	}
 
+	if !t.Required && t.Value == "" {
+		return nil
+	}
```
Comment (abborg) [inline pkg/cli/string_flag.go:90]: This is the core fix. The other flag `Validate` funcs have something like this for the zero value for their types, but it seems this one was skipped. Notably, path_flag.go does have this check, so that suggests to me that this might have been unintentionally missing here.

File: pkg/cli/int_flag.go (hunk 1 above)
Comment (valentindeaconu) [reply pkg/cli/int_flag.go]: This was intentional. It's what makes an explicitly-empty value fail the required validation (e.g. setting `--token-signing-secret=""` exits on master with `required but not set`, while passing on this branch, which is incorrect).

Please revert these. I'll have another look over this package in the near future, hopefully migrating it to generics.

File: pkg/cli/string_flag_test.go [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+		t.Fatal(err)
+	}
+}
+
+func TestStringFlag_Validate_InvalidChoice(t *testing.T) {
+	flag := StringFlag{Choices: []string{"a", "b"}, Value: "z"}
+	if err := flag.Validate(); err == nil {
+		t.Fatal("expected invalid choice")
+	}
+}
```
Comment (valentindeaconu) [inline pkg/cli/string_flag_test.go:196]: Can you add one more test combining those two? Basically an optional flag with multiple choices, with no value provided (empty)? This will be the regression test for this exact fix.

**Review by valentindeaconu — COMMENTED** [review]
Thank you for submitting this. I've left two comments, can you please have a look over them?

File: pkg/cli/string_flag_test.go (hunk 7 above)
Comment (abborg) [reply pkg/cli/string_flag_test.go:196]: Sure thing. I'll revert the explicit-empty case as you requested in the other comment and then add a regression test. Thanks for the fast review!

**Review by valentindeaconu — APPROVED** [review]
Thank you for this work!
